<a href="https://colab.research.google.com/github/GabrielaRC29/DV_23_II/blob/main/Occupationalhealth.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **1. Libraries**



In [2]:
import numpy as np
import pandas as pd
import plotly.express as px

# **2. Import file**

* Upload the file to colab
* Right click on the file and copy the path

In [4]:
# Read the file

## Save the path
p ='/content/OccupationalHealth. (1).xlsx'
## Read and create the dataframe
d = pd.read_excel(p)
d.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 102 entries, 0 to 101
Data columns (total 7 columns):
 #   Column                               Non-Null Count  Dtype 
---  ------                               --------------  ----- 
 0   DATA YEAR                            102 non-null    int64 
 1   ECONOMIC SECTOR                      102 non-null    object
 2   AFFILIATED COMPANIES                 102 non-null    int64 
 3   Dependent affiliated workers         102 non-null    int64 
 4   Independent affiliated workers       102 non-null    int64 
 5   Qualified work accidents             102 non-null    int64 
 6   Diseases classified as occupational  102 non-null    int64 
dtypes: int64(6), object(1)
memory usage: 5.7+ KB


In [5]:
# Explore
d.head()

DATA YEAR         ECONOMIC SECTOR  AFFILIATED COMPANIES  \
0       2017        Domestic service                 53835   
1       2017             Real Estate                 53701   
2       2017                   Trade                 41192   
3       2017  Manufacturing industry                 23517   
4       2017            Construction                 22046   

   Dependent affiliated workers  Independent affiliated workers  \
0                         47527                             232   
1                       1088311                           26525   
2                        388326                           11574   
3                        351434                            6342   
4                        258411                            5902   

   Qualified work accidents  Diseases classified as occupational  
0                       540                                    8  
1                     46859                                  570  
2                     17172                                  275  
3                     24795                                 1248  
4                     15994                                   70

In [6]:
# Explore
d.tail()

DATA YEAR                             ECONOMIC SECTOR  \
97        2022                                     Fishing   
98        2022                            Domestic service   
99        2022     Community, social and personal services   
100       2022                  social and Health Services   
101       2022  Transportation, storage and communications   

     AFFILIATED COMPANIES  Dependent affiliated workers  \
97                     73                           502   
98                  76709                         47163   
99                  12128                        127319   
100                  8981                        159711   
101                 14158                        345336   

     Independent affiliated workers  Qualified work accidents  \
97                                6                        20   
98                              171                       601   
99                           136745                      6034   
100                           53163                     10547   
101                           46102                     14123   

     Diseases classified as occupational  
97                                     1  
98                                    22  
99                                   241  
100                                 8143  
101                                  187

# **3. Cleaning**

# Column names

In [7]:
# Rename
d.rename(columns = {'DATA YEAR':'year'}, inplace = True)
d.rename(columns = {'ECONOMIC SECTOR':'economy'}, inplace = True)
d.rename(columns = {'AFFILIATED COMPANIES':'aff_comp'}, inplace = True)
d.rename(columns = {'Dependent affiliated workers':'workers_dep'}, inplace = True)
d.rename(columns = {'Independent affiliated workers':'workers_ind'}, inplace = True)
d.rename(columns = {'Diseases classified as occupational':'diseases'}, inplace = True)
d.rename(columns = {'Qualified work accidents':'accidents'}, inplace = True)

d.head()

year                 economy  aff_comp  workers_dep  workers_ind  \
0  2017        Domestic service     53835        47527          232   
1  2017             Real Estate     53701      1088311        26525   
2  2017                   Trade     41192       388326        11574   
3  2017  Manufacturing industry     23517       351434         6342   
4  2017            Construction     22046       258411         5902   

   accidents  diseases  
0        540         8  
1      46859       570  
2      17172       275  
3      24795      1248  
4      15994        70

## Replace columns

In [ ]:
d = d.replace('N.A.','Sin dato')

# **4. Indexing**

In [8]:
# Duplicate the dataframe
d2 = d

In [9]:
# Duplicate year as string / How to create a column
d2['year_str'] = d2['year'].astype(str)
d2.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 102 entries, 0 to 101
Data columns (total 8 columns):
 #   Column       Non-Null Count  Dtype 
---  ------       --------------  ----- 
 0   year         102 non-null    int64 
 1   economy      102 non-null    object
 2   aff_comp     102 non-null    int64 
 3   workers_dep  102 non-null    int64 
 4   workers_ind  102 non-null    int64 
 5   accidents    102 non-null    int64 
 6   diseases     102 non-null    int64 
 7   year_str     102 non-null    object
dtypes: int64(6), object(2)
memory usage: 6.5+ KB


In [10]:
# Indexing
## Create the index

idx= pd.MultiIndex.from_product([d2['economy'].unique(),range(d2.year.min(), d2.year.max()+1)],
                                names = ['economy', 'year'])

## Attach the index
d2 = d2.set_index(['economy','year','aff_comp'])

d2.head()

workers_dep  workers_ind  accidents  \
economy                year aff_comp                                        
Domestic service       2017 53835           47527          232        540   
Real Estate            2017 53701         1088311        26525      46859   
Trade                  2017 41192          388326        11574      17172   
Manufacturing industry 2017 23517          351434         6342      24795   
Construction           2017 22046          258411         5902      15994   

                                      diseases year_str  
economy                year aff_comp                     
Domestic service       2017 53835            8     2017  
Real Estate            2017 53701          570     2017  
Trade                  2017 41192          275     2017  
Manufacturing industry 2017 23517         1248     2017  
Construction           2017 22046           70     2017

## **5. Delete NAN**

In [11]:
# Display if there are NaN
d2.isna()

workers_dep  \
economy                                    year aff_comp                
Domestic service                           2017 53835           False   
Real Estate                                2017 53701           False   
Trade                                      2017 41192           False   
Manufacturing industry                     2017 23517           False   
Construction                               2017 22046           False   
...                                                               ...   
Fishing                                    2022 73              False   
Domestic service                           2022 76709           False   
Community, social and personal services    2022 12128           False   
social and Health Services                 2022 8981            False   
Transportation, storage and communications 2022 14158           False   

                                                          workers_ind  \
economy                                    year aff_comp                
Domestic service                           2017 53835           False   
Real Estate                                2017 53701           False   
Trade                                      2017 41192           False   
Manufacturing industry                     2017 23517           False   
Construction                               2017 22046           False   
...                                                               ...   
Fishing                                    2022 73              False   
Domestic service                           2022 76709           False   
Community, social and personal services    2022 12128           False   
social and Health Services                 2022 8981            False   
Transportation, storage and communications 2022 14158           False   

                                                          accidents  diseases  \
economy                                    year aff_comp                        
Domestic service                           2017 53835         False     False   
Real Estate                                2017 53701         False     False   
Trade                                      2017 41192         False     False   
Manufacturing industry                     2017 23517         False     False   
Construction                               2017 22046         False     False   
...                                                             ...       ...   
Fishing                                    2022 73            False     False   
Domestic service                           2022 76709         False     False   
Community, social and personal services    2022 12128         False     False   
social and Health Services                 2022 8981          False     False   
Transportation, storage and communications 2022 14158         False     False   

                                                          year_str  
economy                                    year aff_comp            
Domestic service                           2017 53835        False  
Real Estate                                2017 53701        False  
Trade                                      2017 41192        False  
Manufacturing industry                     2017 23517        False  
Construction                               2017 22046        False  
...                                                            ...  
Fishing                                    2022 73           False  
Domestic service                           2022 76709        False  
Community, social and personal services    2022 12128        False  
social and Health Services                 2022 8981         False  
Transportation, storage and communications 2022 14158        False  

[102 rows x 5 columns]

In [12]:
# Delete NaN
d2 = d2.dropna()
d2.info()

<class 'pandas.core.frame.DataFrame'>
MultiIndex: 102 entries, ('Domestic service', 2017, 53835) to ('Transportation, storage and communications', 2022, 14158)
Data columns (total 5 columns):
 #   Column       Non-Null Count  Dtype 
---  ------       --------------  ----- 
 0   workers_dep  102 non-null    int64 
 1   workers_ind  102 non-null    int64 
 2   accidents    102 non-null    int64 
 3   diseases     102 non-null    int64 
 4   year_str     102 non-null    object
dtypes: int64(4), object(1)
memory usage: 8.2+ KB


## **6. Descriptive statistics**

In [13]:
# Basic
d2[['workers_ind','diseases','accidents']].describe().applymap(lambda x: "{:.1f}".format(x))


workers_ind diseases accidents
count       102.0    102.0     102.0
mean      19469.1    526.4    9455.8
std       27864.1   1532.3   10603.8
min           0.0      0.0      10.0
25%         709.0     54.5    1831.2
50%        7207.0    175.0    6293.0
75%       36684.0    302.8   14324.8
max      136745.0  10639.0   46859.0

**Interpretation**

The information provided by the code shows us that :

Within sectors of the economy, there is an average of 19,469 people who are independently affiliated with an insurance company. Likewise, there is an average of 526 deaths and 9,455 accidents reported as occupational by these workers.
The economy with the most accidents reported 46,859 of these; The economy with the most deaths reported 10,639 of these.
The standard deviation of the data is high, which means that there is great variability between workers in each economy, their accidents and deaths respectively.



In [14]:
# By index
d2[['workers_ind']].groupby('economy').describe().applymap(lambda x: "{:.1f}".format(x))

workers_ind                    \
                                                   count     mean      std   
economy                                                                      
Agriculture, Livestock, Hunting and Forestry         6.0  19597.5  32261.7   
Community, social and personal services              6.0  50559.8  58341.2   
Construction                                         6.0   7793.5   1775.9   
Domestic service                                     6.0    253.3    143.9   
Education                                            6.0  44358.0   7664.9   
Electric, gas and water                              6.0    449.2    238.3   
Extraterritorial Bodies                              4.0    281.0    230.6   
Financial                                            6.0   9116.8   3768.7   
Fishing                                              6.0      5.5      2.9   
Hotels and restaurants                               6.0   1391.7    466.1   
Manufacturing industry                               6.0   6291.8   2587.2   
Mines and quarries                                   6.0   1332.8    622.6   
Organos Extraterritoriales                           2.0    318.0      8.5   
Public Administration and Defense                    6.0  71274.8  27422.8   
Real Estate                                          6.0  31997.2  14984.4   
Trade                                                6.0   7811.8   3767.0   
Transportation, storage and communications           6.0  36610.7  19658.3   
social and Health Services                           6.0  41837.3  17012.8   

                                                                         \
                                                  min      25%      50%   
economy                                                                   
Agriculture, Livestock, Hunting and Forestry    159.0   4177.2   8832.0   
Community, social and personal services        1852.0  15737.0  18311.0   
Construction                                   5900.0   6126.5   8065.0   
Domestic service                                 17.0    186.2    282.0   
Education                                     35658.0  38803.0  43859.0   
Electric, gas and water                          79.0    323.8    503.0   
Extraterritorial Bodies                          17.0    155.0    272.0   
Financial                                      1491.0   9933.8  10662.5   
Fishing                                           0.0      5.2      6.5   
Hotels and restaurants                         1020.0   1067.0   1180.0   
Manufacturing industry                         1553.0   6178.5   6638.5   
Mines and quarries                              131.0   1314.5   1506.0   
Organos Extraterritoriales                      312.0    315.0    318.0   
Public Administration and Defense             20250.0  68541.5  76402.5   
Real Estate                                    3743.0  29514.5  39272.0   
Trade                                          1031.0   6975.5   8489.5   
Transportation, storage and communications      695.0  32463.8  41564.0   
social and Health Services                     8187.0  44593.8  45528.0   

                                                                 
                                                  75%       max  
economy                                                          
Agriculture, Livestock, Hunting and Forestry  11425.8   84846.0  
Community, social and personal services       89523.5  136745.0  
Construction                                   9364.5    9453.0  
Domestic service                                367.2     389.0  
Education                                     47750.2   56607.0  
Electric, gas and water                         572.0     751.0  
Extraterritorial Bodies                         398.0     563.0  
Financial                                     10980.2   11108.0  
Fishing                                           7.0       8.0  
Hotels and restaurants                

**Interpretation**

The information provided by the code shows us that :

Occupational Health data takes into account 18 sectors of the economy; in which there is a sample of 6 in most industries. The economy with the highest average number of independently affiliated workers is Defense and Public Administration.
By 2022, it will present the largest number of independent affiliates with 98,110 workers.The standard deviation for the industry with the largest number of workers represents that there is great variability between its data between the years 2017 and 2022.
While on the other hand, the economy with the average number of independent affiliated workers is fishing.










##**7. Basic graph**

In [15]:
#Scatter matrix
fig1 = px.scatter_matrix(d2, dimensions = ['workers_ind', 'accidents','diseases'], color = 'year_str')
fig1.show()



In [16]:
#Scatter matrix
fig1 = px.scatter_matrix(d2, dimensions = ['workers_ind', 'accidents','diseases'], color = 'year_str')
fig1.update_traces(diagonal_visible = False)
fig1.update_traces(showupperhalf = False)
fig1.show()

**Intepretation**

Occupational health is one of the most important rights of workers.
The graphs show a significant concentration of occupational accidents that inevitably exceed the number of occupational deaths.A hypothesis could be created that the economy with the greatest number of workers is also the industry with the greatest number of illnesses and accidents.
This information is of great use for affiliate companies, as they also know what type of risks they must manage and in what quantity depending on the industry.

## **8. Pivots**

In [17]:
Education = d2.filter(like = 'Education', axis=0)
Education.head()

workers_dep  workers_ind  accidents  diseases  \
economy   year aff_comp                                                  
Education 2017 4554           229920        45324       5702       123   
          2018 4680           229851        48559       5569       140   
          2019 4481           205588        37606       5986       117   
          2020 4512           193897        56607       4024       320   
          2021 4701           185340        35658       3308       166   

                        year_str  
economy   year aff_comp           
Education 2017 4554         2017  
          2018 4680         2018  
          2019 4481         2019  
          2020 4512         2020  
          2021 4701         2021

In [18]:
y21 = d2.filter(like = '2021', axis=0)
y21.head()

workers_dep  \
economy                                      year aff_comp                
Public Administration and Defense            2021 1468           167679   
Agriculture, Livestock, Hunting and Forestry 2021 8522            62404   
Trade                                        2021 50243          415128   
Construction                                 2021 28012          264136   
Education                                    2021 4701           185340   

                                                            workers_ind  \
economy                                      year aff_comp                
Public Administration and Defense            2021 1468            90152   
Agriculture, Livestock, Hunting and Forestry 2021 8522             9910   
Trade                                        2021 50243            9231   
Construction                                 2021 28012            9330   
Education                                    2021 4701            35658   

                                                            accidents  \
economy                                      year aff_comp              
Public Administration and Defense            2021 1468           3896   
Agriculture, Livestock, Hunting and Forestry 2021 8522           7519   
Trade                                        2021 50243         14392   
Construction                                 2021 28012         16199   
Education                                    2021 4701           3308   

                                                            diseases year_str  
economy                                      year aff_comp                     
Public Administration and Defense            2021 1468           495     2021  
Agriculture, Livestock, Hunting and Forestry 2021 8522           302     2021  
Trade                                        2021 50243          357     2021  
Construction                                 2021 28012          127     2021  
Education                                    2021 4701           166     2021

**Pivots**

In [19]:
tmp = d2.filter(like ='Domestic service', axis=0)
tmp = tmp.reset_index()
tmp.head()

pivot_pop = tmp.pivot(index = "year",
                    columns = "economy",
                    values = "workers_ind")
pivot_pop.head()

economy  Domestic service
year                     
2017                  232
2018                  379
2019                  389
2020                   17
2021                  332

In [20]:
domestic = d2.filter(like = 'Domestic service', axis=0)
domestic.head()

workers_dep  workers_ind  accidents  diseases  \
economy          year aff_comp                                                  
Domestic service 2017 53835           47527          232        540         8   
                 2018 59609           48194          379        626        28   
                 2019 63563           49238          389        707        11   
                 2020 66174           47104           17        605        35   
                 2021 77297           45624          332        631        30   

                               year_str  
economy          year aff_comp           
Domestic service 2017 53835        2017  
                 2018 59609        2018  
                 2019 63563        2019  
                 2020 66174        2020  
                 2021 77297        2021

In [21]:
fig2 = px.bar(domestic, x='year_str', y='workers_ind')

# Without the Legend

fig2.update_layout(showlegend = False)

# Center the title
fig2.update_layout(title_text = 'Domestic service workers',
                  title_font_size = 25,
                  title_x = 0.5)

# Modify the text in the axis
fig2.update_xaxes(title = 'Years', title_font_size = 20,
                 dtick= 'M1',
                 tickformat = '%b\n%Y')
fig2.update_yaxes(title = 'Workers', title_font_size = 20)
fig2.show()

# **Interpretation**

For this case, the domestic service economy was chosen, it is intended to represent that the number of independently affiliated workers in this sector had its highest record in 2019, having a next and important drop in 2020 of almost 95%. . But curiously, for the year 2021, the recovery was prompt, so the causal factors such as greater employability in this sector would have to be evaluated.



#**9. Another Graphs**

## Double axis

In [22]:
## Libraries
import plotly.graph_objects as go
from plotly.subplots import make_subplots

In [23]:
## Create una figuera whit secondary axis

fig3 = make_subplots(specs=[[{"secondary_y": True}]])

## add each graph
fig3.add_trace(
    go.Scatter(x=domestic.year_str, y=domestic.workers_ind, name='workers_ind'),
    secondary_y=False,
)
fig3.add_trace(
    go.Scatter(x=domestic.year_str, y=domestic['diseases'], name='diseases'),
    secondary_y=True,
)

# Without the Legend

fig3.update_layout(showlegend = False)

# Center the title
fig3.update_layout(title_text = 'Deaths in domestic service',
                  title_font_size = 25,
                  title_x = 0.5)

# Modify the text in the axis
fig3.update_xaxes(title = 'Years', title_font_size = 18,
                 dtick= 'M1',
                 tickformat = '%b\n%Y')
fig3.update_yaxes(title = 'Workers', title_font_size = 18)

fig3.update_yaxes(title='Diseases', title_font_size= 18, secondary_y=True)

fig3.show()



**Interpretation**

The domestic service sector has had many regulations in recent years, so I find it interesting to analyze the number of deaths that independently affiliated workers have.
Particularly for the year with the highest registration of workers, the number of deaths was the lowest with 11 of these, while the year with the lowest registration of workers recorded the highest level of occupational deaths with 35 of them. It is important to review the causes of deaths in a sector like this, given that it is one of the sectors with the largest number of affiliated companies.

